File: solution/49. Group Anagrams/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<int>> m;
        
        for (int i = 0; i < strs.size(); ++i) {
            string key = strs[i];
            sort(key.begin(), key.end());
            m[key].push_back(i);
        }
        
        for (auto& kv : m) {
            ans.push_back({});
            for (int i : kv.second)
                ans.back().push_back(strs[i]);
        }
        return ans;
    }
};

File: solution/49. Group Anagrams/49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = defaultdict(list)
        
        for word in strs:
            key = ''.join(sorted(word))
            m[key].append(word)
            
        return list(m.values())
    
